bugbug/models/component.py

```diff
@@ -203,3 +204,42 @@ def is_meaningful(product, component):
 
     def get_feature_names(self):
         return self.extraction_pipeline.named_steps["union"].get_feature_names()
+
+    def check(self):
+        success = super().check()
+
+        # Get the number of bugs per full component to fasten up the check
+        bugs_number = get_product_component_count()
+
+        # Check that the most meaningful product components stills have at
+        # least a bug in this component. If the check is failing that could
+        # means that:
+        # - A component has been renamed / removed
+        # - TODO: Complete this list
+
+        for product, component in self.meaningful_product_components:
+            full_comp = f"{product}::{component}"
+
+            if full_comp not in bugs_number.keys():
+                msg = f"Component {component!r} of product {product!r} have 0 bugs in it, failure"
+                print(msg)
+                success = False
+
+        # Check that the conflated components still exists and have at least
+        # one bug
+
+        # Assert all conflated components are either in conflated_components_mapping or exist as components.
+        for conflated_component in self.CONFLATED_COMPONENTS:
+
+            matching_components = [
+                full_comp
+                for full_comp in bugs_number.keys()
+                if full_comp.startswith(conflated_component)
+            ]
+
+            if not matching_components:
+                msg = f"It should be possible to maps {conflated_component}"
+                print(msg)
+                success = False
+
```

Comment:
We should also do the checks https://github.com/mozilla/bugbug/blob/6937e0e5e860b097e71bac95e0ab941202f43dc1/bugbug/models/component.py#L184 and https://github.com/mozilla/bugbug/blob/6937e0e5e860b097e71bac95e0ab941202f43dc1/bugbug/models/component.py#L191